go/vt/mysqlctl/mysqlshellbackupengine.go

```diff
@@ -184,6 +190,7 @@ func (be *MySQLShellBackupEngine) ExecuteBackup(ctx context.Context, params Back
 
 	// Get exit status.
 	if err := cmd.Wait(); err != nil {
+		pipeWriter.Close() // make sure we close the writer so the goroutines above will complete.
```

Comment:
we want to call `Close()` on it before `cmdWg.Wait()` on line 199 since that will cause the Scanner on the `scanLinesToLogger()` goroutines we set above to hit and EOF and close gracefully. if we defer the `Close()` then likely `cmdWg.Wait()` will hang forever since the pipe is still open.

but in case we hit an execution error, we still need to close it to avoid those goroutines to leak (something that was happening before but it only noticed with the new tests)

Reply:
Ah, I see where we close it on line 198 now. Didn't see that before. 👍